src/aosm/azext_aosm/deploy/deploy_with_arm.py

```diff
@@ -0,0 +1,376 @@
+# --------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT
+# License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------
+"""Contains class for deploying generated definitions using the Python SDK."""
+import json
+import os
+import shutil
+import subprocess  # noqa
+from typing import Any, Dict, Optional
+
+from knack.log import get_logger
+from azext_aosm.deploy.artifact_manifest import ArtifactManifestOperator
+from azext_aosm.util.management_clients import ApiClients
+from azure.mgmt.resource.resources.v2021_04_01.models import DeploymentExtended
+from pathlib import Path
+
+from azext_aosm.deploy.pre_deploy import PreDeployerViaSDK
```

Comment:
The naming here confuses me. From what I can tell you've moved the common function between the two deployment approaches out into a separate file to use with both of them (good) but the names of things still seem to keep mentioning SDK specifically.

Reply:
I've added a comment to DeployerViaArm to hopefully distinguish this and actually deleted deploy_with_sdk.py for now because it was unused and not hard to write again.